File: packages/shared-types/src/notifications.ts
/**
 * Push notification types.
 *
 * Token registration:
 *   POST /notifications/token → PushTokenResponse
 *   DELETE /notifications/token/{id}
 *
 * Notification events are server-scheduled; the mobile side only
 * needs to register a token and handle incoming push payloads.
 */

import type { ISODateString, UUID } from './common';

export type PushPlatform = 'ios' | 'android' | 'web';

export type NotificationType =
  | 'proposal_received'
  | 'booking_confirmed'
  | 'booking_declined'
  | 'booking_cancelled'
  | 'reminder';

export interface RegisterPushTokenRequest {
  token: string;
  platform: PushPlatform;
}

export interface PushTokenResponse {
  id: UUID;
  userId: UUID;
  token: string;
  platform: PushPlatform;
  createdAt: ISODateString;
}

/** Payload included in push notification data field. */
export interface PushNotificationData {
  type: NotificationType;
  bookingId?: UUID;
}

File: apps/api/app/schemas/notifications.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


class RegisterPushTokenRequest(BaseModel):
    token: str = Field(..., min_length=1, max_length=256)
    platform: Literal["ios", "android", "web"] = "ios"


class PushTokenResponse(BaseModel):
    id: UUID
    user_id: UUID
    token: str
    platform: str
    created_at: datetime

    model_config = {"from_attributes": True}


class ProcessNotificationsResult(BaseModel):
    processed: int
    failed: int
